### PR TITLE
add addon manager

### PR DESCRIPTION
**Pull request by haoqing0110**

### CODE REVIEW

File: operator/v1/types.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ type ClusterManagerSpec struct {
 	// +kubebuilder:default=quay.io/open-cluster-management/placement
 	PlacementImagePullSpec string `json:"placementImagePullSpec,omitempty"`
 
+	// AddOnManagerImagePullSpec represents the desired image configuration of addon manager controller/webhook installed on hub.
+	// +optional
+	// +kubebuilder:default=quay.io/open-cluster-management/addon-manager
+	AddOnManagerImagePullSpec string `json:"addOnManagerImagePullSpec,omitempty"`
+
```
Comment (zhujian7) [inline operator/v1/types.go:51]:
```suggestion
	// AddonManagerConfiguration ...
	AddonManagerConfiguration *AddonManagerConfiguration `json:"addonManagerConfiguration,omitempty"`
}

type AddonManagerConfiguration struct {
	// Enabled ...
	Enabled bool `json:"enabled"`

	// AddOnManagerImagePullSpec represents the desired image configuration of addon manager controller/webhook installed on hub.
	// +optional
	// +kubebuilder:default=quay.io/open-cluster-management/addon-manager
	AddOnManagerImagePullSpec string `json:"addOnManagerImagePullSpec,omitempty"`
}

do we need a feature gate to control whether enable/disable this? how about 👆🏻？
```

File: operator/v1/types.go [hunk 2]
```diff
@@ -87,6 +96,34 @@ type WorkConfiguration struct {
...
+type AddOnMangerConfiguration struct {
+	// Mode is either Enable, Disable, "" where "" is Disable by default.
+	// In Enable mode, the component will be installed.
+	// In Disable mode, the component will not be installed.
+	// +kubebuilder:default:=Disable
+	// +kubebuilder:validation:Enum:=Enable;Disable
+	// +optional
+	Mode ComponentModeType `json:"mode,omitempty"`
+
+	// FeatureGates represents the list of feature gates for work
```
Comment (qiujian16) [inline operator/v1/types.go]: for addon manager

File: operator/v1/types.go [hunk 3]
```diff
@@ -61,6 +66,11 @@ type ClusterManagerSpec struct {
 	// WorkConfiguration contains the configuration of work
 	// +optional
 	WorkConfiguration *WorkConfiguration `json:"workConfiguration,omitempty"`
+
+	// AddOnManagerConfiguration contains the configuration of addon manager
+	// +optional
+	// +kubebuilder:default={mode: Disable}
+	AddOnManagerConfiguration AddOnManagerConfiguration `json:"addOnManagerConfiguration,omitempty"`
```
Comment (qiujian16) [inline operator/v1/types.go]: should this be a pointer?

Comment (haoqing0110) [reply operator/v1/types.go]: fixed, pointer should be better.

**Pull request merged**